core/src/main/java/org/axonframework/commandhandling/model/inspection/AnnotatedChildEntity.java

```diff
@@ -71,7 +73,9 @@ public void publish(EventMessage<?> msg, P declaringInstance) {
         if (forwardEvents) {
             Iterable<C> targets = eventTargetResolver.apply(msg, declaringInstance);
             if (targets != null) {
-                targets.forEach(target -> this.entityModel.publish(msg, target));
+                Collection<C> copy = new ArrayList<>();
```

Comment:
I added detection of the targets implementation to determine if and how to copy it before iterating.
Unfortunately I couldn't come up with a test that contains an Iterable which is not a Collection, so that part should be checked theoretically I geuss.